File: tutor-backend/src/main/java/com/tutor/plan/model/LearningPlan.java
package com.tutor.plan.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "learning_plans")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class LearningPlan {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private Long goalId;

    @Column(nullable = false)
    private Long userId;

    @Column(nullable = false)
    private String title;

    @Column(length = 1000)
    private String description;

    @Builder.Default
    private Integer totalMilestones = 0;

    @Builder.Default
    private Integer completedMilestones = 0;

    @Builder.Default
    private String status = "ACTIVE"; // ACTIVE, ADAPTED, COMPLETED

    @OneToMany(mappedBy = "learningPlan", cascade = CascadeType.ALL, orphanRemoval = true)
    @Builder.Default
    private List<Milestone> milestones = new ArrayList<>();

    @Builder.Default
    private LocalDateTime createdAt = LocalDateTime.now();

    @Builder.Default
    private LocalDateTime updatedAt = LocalDateTime.now();

    public void addMilestone(Milestone milestone) {
        milestones.add(milestone);
        milestone.setLearningPlan(this);
    }
}

File: tutor-frontend/src/types/LearningPlan.ts
export interface Task {
  id: number;
  sequenceOrder: number;
  title: string;
  resourceUrl?: string;
  durationMinutes: number;
  status: 'PENDING' | 'IN_PROGRESS' | 'COMPLETED';
}

export interface Milestone {
  id: number;
  sequenceOrder: number;
  title: string;
  description: string;
  estimatedHours: number;
  isCompleted: boolean;
  tasks: Task[];
}

export interface LearningPlan {
  id: number;
  goalId: number;
  userId: number;
  title: string;
  description: string;
  totalMilestones: number;
  completedMilestones: number;
  progressPercentage: number;
  status: 'ACTIVE' | 'ADAPTED' | 'COMPLETED';
  milestones: Milestone[];
  createdAt: string;
  updatedAt: string;
}
